lib/resty/core/ndk.lua

```diff
@@ -0,0 +1,90 @@
+-- Copyright (C) by OpenResty Inc.
+
+
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+
```

Comment:
We should only allow loading this module in the http subsystem via the call `base.allows_subsystem('http')`.